sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysisResult.java

```diff
@@ -6,86 +6,85 @@
+ *     Composed of a list of sensitivity values in pre-contingency and post-contingency states.
  * </p>
  *
  * A single sensitivity analysis should return, besides its status and some stats on the
- * analysis itself, all the sensitivity values for each factor (combination of a monitoredBranch and a specific
- * equipment or group of equipments). The HADES2 sensitivity provider used with Powsybl offers the
- * possibility to calculate the sensitivity on a set of contingencies besides the N state.
- * The analysis is launched only once, but the solver itself
- * modifies the matrix for each state of the network to output a full set of results.
- * In the sensitivity API, it has been allowed to provide a list of contingencies as an optional input,
+ * analysis itself, all the sensitivity values for each factor (combination of a monitored equipment or bus and a specific
+ * equipment or group of equipments). The chosen sensitivity provider offers the possibility to calculate the sensitivity
+ * on a set of contingencies besides the pre-contingency state.
+ * Note that the analysis is launched only once, but the solver itself modifies the matrix for each state of the network
+ * to output a full set of results. In the sensitivity API, it has been allowed to provide a list of contingencies as an input,
  * which then triggers such a sensitivity analysis.
  * The full set of results consists of :
- *  - the list of sensitivity values in N
- *  - the lists of sensitivity values for each N-1 situation
+ *  - the list of sensitivity values in pre-contingency and post-contingency states.
  *  - some metadata (status, stats, logs)
  *
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  * @see SensitivityValue
  */
 public class SensitivityAnalysisResult {
 
-    private static final String VALUE_NOT_FOUND = "Sensitivity value not found for function %s and variable %s.";
-    private static final String VALUE_NOT_FOUND_CONTINGENCY = "Sensitivity value not found for function %s and variable %s at contingencyId %s.";
+    private static final Logger LOGGER = LoggerFactory.getLogger(SensitivityAnalysisResult.class);
 
-    @JsonProperty("ok")
     private final boolean ok;
 
-    @JsonProperty("metrics")
     private final Map<String, String> metrics;
 
-    @JsonProperty("logs")
     private final String logs;
 
-    @JsonProperty("values")
-    private final List<SensitivityValue> sensitivityValues;
+    private final List<SensitivityValue> values;
+
+    private Map<String, List<SensitivityValue>> valuesByContingencyId = new HashMap<>();
 
-    @JsonProperty("contingenciesValues")
-    private final Map<String, List<SensitivityValue>> sensitivityValuesContingencies;
+    private Map<Triple<String, String, String>, SensitivityValue> valuesByContingencyIdAndFunctionIdAndVariableId = new HashMap<>();
```

Comment:
Would it be possible to have two maps:
- one containing the sensitivity values by contingency, function and variable 
- one containing the function reference values by contingency and function only (and remove the function reference from the SensitivityValue object)

This could help make sensitivity analysis providers more robust, and also avoid duplicating the same information many times.

Reply:
You mean by contingency and variable only?

Reply:
I do mean by contingency and function. For instance, if we have 5 contingencies, 10 branches (functions) and 3 PSTs (variables), we would have:
- one map containing the sensitivity values of each pst on each branch for each contingency
- one map containing the flow of each branch for each contingency